internal/am/jobs.go

```diff
@@ -0,0 +1,120 @@
+package am
+
+import (
+	context "context"
+	"database/sql"
+
+	"github.com/jonboulle/clockwork"
+	"go.artefactual.dev/amclient"
+
+	"github.com/artefactual-sdps/enduro/internal/package_"
+)
+
+var jobStatusToPreservationTaskStatus = map[amclient.JobStatus]package_.PreservationTaskStatus{
+	amclient.JobStatusUnknown:    package_.TaskStatusUnspecified,
+	amclient.JobStatusComplete:   package_.TaskStatusDone,
+	amclient.JobStatusProcessing: package_.TaskStatusInProgress,
+	amclient.JobStatusFailed:     package_.TaskStatusError,
+}
+
+type PreservationAction struct {
+	// clock is a service that provides clock time.
+	clock  clockwork.Clock
+	jobSvc amclient.JobsService
+	pkgSvc package_.Service
+
+	// id is the Preservation Action ID.
+	id uint
+
+	// savedIDs caches the ID of jobs that have already been saved so we don't
+	// create duplicates.
+	savedIDs map[string]struct{}
```

Comment:
This is perfect, but now wondering whether we should eventually adopt some kind of generic data structure, e.g. https://github.com/zyedidia/generic has a set type - see https://go.dev/play/p/hHNbIN2msXf. `struct{}{}` is a memory trick that may be as well hidden I guess.

Reply:
I meant in the long term really, not asking you to do it now.